### PR TITLE
HBASE-26027 The calling of HTable.batch blocked at AsyncRequestFuture…

### PR DESCRIPTION
**Pull request by bsglz**
…Impl.waitUntilDone caused by ArrayStoreException

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 32s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  7s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ branch-2 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 40s |  branch-2 passed  |
| +1 :green_heart: |  compile  |   0m 28s |  branch-2 passed  |
| +1 :green_heart: |  shadedjars  |   6m  2s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 26s |  branch-2 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 19s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 26s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 26s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   6m  6s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 24s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m 46s |  hbase-client in the patch passed.  |
|  |   |  25m 35s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3419/1/artifact/yetus-jdk8-hadoop2-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3419 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux d736878b56c2 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2 / e3eb760864 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3419/1/testReport/ |
| Max. process+thread count | 343 (vs. ulimit of 12500) |
| modules | C: hbase-client U: hbase-client |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3419/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 41s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  7s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ branch-2 Compile Tests _ |
| -1 :x: |  mvninstall  |   4m  9s |  root in branch-2 failed.  |
| +1 :green_heart: |  compile  |   0m 32s |  branch-2 passed  |
| +1 :green_heart: |  shadedjars  |   6m 51s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 30s |  branch-2 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 58s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 31s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 31s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   6m 50s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 28s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m 22s |  hbase-client in the patch passed.  |
|  |   |  28m 14s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3419/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3419 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 785986e68faa 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2 / e3eb760864 |
| Default Java | AdoptOpenJDK-11.0.10+9 |
| mvninstall | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3419/1/artifact/yetus-jdk11-hadoop3-check/output/branch-mvninstall-root.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3419/1/testReport/ |
| Max. process+thread count | 282 (vs. ulimit of 12500) |
| modules | C: hbase-client U: hbase-client |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3419/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  9s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ branch-2 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 40s |  branch-2 passed  |
| +1 :green_heart: |  compile  |   1m  6s |  branch-2 passed  |
| +1 :green_heart: |  checkstyle  |   0m 34s |  branch-2 passed  |
| +1 :green_heart: |  spotbugs  |   1m 12s |  branch-2 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 20s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  3s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m  3s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 34s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  11m 46s |  Patch does not cause any errors with Hadoop 3.1.2 3.2.1.  |
| +1 :green_heart: |  spotbugs  |   1m 19s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 15s |  The patch does not generate ASF License warnings.  |
|  |   |  33m 31s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3419/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3419 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 2f349c1e3848 4.15.0-65-generic #74-Ubuntu SMP Tue Sep 17 17:06:04 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2 / e3eb760864 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| Max. process+thread count | 96 (vs. ulimit of 12500) |
| modules | C: hbase-client U: hbase-client |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3419/1/console |
| versions | git=2.17.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 34s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  5s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ branch-2 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m 20s |  branch-2 passed  |
| +1 :green_heart: |  compile  |   0m 32s |  branch-2 passed  |
| +1 :green_heart: |  shadedjars  |   6m 50s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 31s |  branch-2 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m  0s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 32s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 32s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   6m 47s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 28s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m 49s |  hbase-client in the patch passed.  |
|  |   |  28m 29s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3419/2/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3419 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 978d5dd5e2d9 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2 / 7e6b66c867 |
| Default Java | AdoptOpenJDK-11.0.10+9 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3419/2/testReport/ |
| Max. process+thread count | 300 (vs. ulimit of 12500) |
| modules | C: hbase-client U: hbase-client |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3419/2/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 27s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  5s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ branch-2 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m 24s |  branch-2 passed  |
| +1 :green_heart: |  compile  |   0m 29s |  branch-2 passed  |
| +1 :green_heart: |  shadedjars  |   6m 46s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 27s |  branch-2 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 47s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 28s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 28s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   6m 37s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 26s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m 51s |  hbase-client in the patch passed.  |
|  |   |  28m 48s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3419/2/artifact/yetus-jdk8-hadoop2-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3419 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux bcd49c28cf8d 4.15.0-136-generic #140-Ubuntu SMP Thu Jan 28 05:20:47 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2 / 7e6b66c867 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3419/2/testReport/ |
| Max. process+thread count | 238 (vs. ulimit of 12500) |
| modules | C: hbase-client U: hbase-client |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3419/2/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 23s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ branch-2 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 46s |  branch-2 passed  |
| +1 :green_heart: |  compile  |   1m  4s |  branch-2 passed  |
| +1 :green_heart: |  checkstyle  |   0m 34s |  branch-2 passed  |
| +1 :green_heart: |  spotbugs  |   1m 12s |  branch-2 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 26s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  2s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m  2s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 32s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  12m 10s |  Patch does not cause any errors with Hadoop 3.1.2 3.2.1.  |
| +1 :green_heart: |  spotbugs  |   1m 21s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 15s |  The patch does not generate ASF License warnings.  |
|  |   |  34m 14s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3419/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3419 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 361bb0e7b14d 4.15.0-65-generic #74-Ubuntu SMP Tue Sep 17 17:06:04 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2 / 7e6b66c867 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| Max. process+thread count | 96 (vs. ulimit of 12500) |
| modules | C: hbase-client U: hbase-client |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3419/2/console |
| versions | git=2.17.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncRequestFutureImpl.java [hunk 1]
```diff
@@ -1140,7 +1140,8 @@ private String buildDetailedErrorMsg(String string, int index) {
   @Override
   public void waitUntilDone() throws InterruptedIOException {
     try {
-      waitUntilDone(Long.MAX_VALUE);
+      long cutoff = (EnvironmentEdgeManager.currentTime() + this.operationTimeout) * 1000L;
```
Comment (Apache9) [inline hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncRequestFutureImpl.java:1143]: What if operationTimeout here is negative(which means no timeout), or Long.MAX_VALUE?

**Review by Apache9 — COMMENTED** [review]
I'm not sure how could this change solve the problem described in the issue, but FWIW, I think we should honor operation timeout.

File: hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncRequestFutureImpl.java (hunk 1 above)
Comment (bsglz) [reply hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncRequestFutureImpl.java:1143]: > What if operationTimeout here is negative(which means no timeout), or Long.MAX_VALUE?

Good point, exclude that cases seems better, will fix later, thanks.

Comment (bsglz) [reply hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncRequestFutureImpl.java:1143]: The type of operationTimeout is int, so seems no need to consider Long.MAX_VALUE case.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 36s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  6s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ branch-2 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 31s |  branch-2 passed  |
| +1 :green_heart: |  compile  |   0m 24s |  branch-2 passed  |
| +1 :green_heart: |  shadedjars  |   5m 56s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 24s |  branch-2 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 21s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 29s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 29s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   6m  9s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 25s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m 45s |  hbase-client in the patch passed.  |
|  |   |  25m 20s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3419/3/artifact/yetus-jdk8-hadoop2-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3419 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 22bfa111e70e 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2 / 166becfd66 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3419/3/testReport/ |
| Max. process+thread count | 346 (vs. ulimit of 12500) |
| modules | C: hbase-client U: hbase-client |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3419/3/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 40s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ branch-2 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 54s |  branch-2 passed  |
| +1 :green_heart: |  compile  |   1m  5s |  branch-2 passed  |
| +1 :green_heart: |  checkstyle  |   0m 35s |  branch-2 passed  |
| +1 :green_heart: |  spotbugs  |   1m 11s |  branch-2 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m 23s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  4s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m  4s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 30s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  11m 51s |  Patch does not cause any errors with Hadoop 3.1.2 3.2.1.  |
| +1 :green_heart: |  spotbugs  |   1m 20s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 14s |  The patch does not generate ASF License warnings.  |
|  |   |  33m 20s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3419/3/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3419 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti checkstyle compile |
| uname | Linux c7bf791d4966 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2 / 166becfd66 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| Max. process+thread count | 96 (vs. ulimit of 12500) |
| modules | C: hbase-client U: hbase-client |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3419/3/console |
| versions | git=2.17.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   6m 25s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  7s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ branch-2 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m 46s |  branch-2 passed  |
| +1 :green_heart: |  compile  |   0m 31s |  branch-2 passed  |
| +1 :green_heart: |  shadedjars  |   7m 36s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 31s |  branch-2 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   4m 30s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 33s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 33s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   7m 36s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 28s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m 54s |  hbase-client in the patch passed.  |
|  |   |  37m  6s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3419/3/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/3419 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux d1b9d2a1b3db 4.15.0-128-generic #131-Ubuntu SMP Wed Dec 9 06:57:35 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2 / 166becfd66 |
| Default Java | AdoptOpenJDK-11.0.10+9 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3419/3/testReport/ |
| Max. process+thread count | 212 (vs. ulimit of 12500) |
| modules | C: hbase-client U: hbase-client |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-3419/3/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Review by Apache9 — APPROVED** [review]

**Pull request merged**